open-sphere-plugins/feature-actions/src/main/java/io/opensphere/featureactions/controller/FeatureActionStateController.java

```diff
@@ -0,0 +1,260 @@
+import io.opensphere.core.modulestate.AbstractLayerStateController;
+import io.opensphere.core.modulestate.ModuleStateController;
+import io.opensphere.core.modulestate.StateUtilities;
+import io.opensphere.core.modulestate.StateXML;
+import io.opensphere.core.util.ColorUtilities;
+import io.opensphere.core.util.XMLUtilities;
+import io.opensphere.core.util.collections.New;
+import io.opensphere.core.util.lang.Pair;
+import io.opensphere.featureactions.model.Action;
+import io.opensphere.featureactions.model.FeatureAction;
+import io.opensphere.featureactions.model.StyleAction;
+import io.opensphere.featureactions.registry.FeatureActionsRegistry;
+import io.opensphere.filterbuilder.filter.v1.Filter;
+import io.opensphere.filterbuilder.impl.WFS100FilterToDataFilterConverter;
+import io.opensphere.mantle.controller.DataGroupController;
+import io.opensphere.mantle.data.DataTypeInfo;
+import io.opensphere.mantle.data.element.mdfilter.FilterException;
+import io.opensphere.mantle.data.element.mdfilter.FilterToWFS100Converter;
+import javafx.collections.ObservableList;
+
+/**
+ * A state controller for {@link FeatureAction}.
+ */
+public class FeatureActionStateController extends AbstractLayerStateController<Pair<FeatureAction, String>>
+{
+    /** Logger reference. */
+    private static final Logger LOGGER = Logger.getLogger(FeatureActionStateController.class);
+
+    /** The registry for feature actions. */
+    private final FeatureActionsRegistry myRegistry;
+
+    /** The data group controller to access layers. */
+    private final DataGroupController myDataGroupController;
+
+    /**
+     * Constructor.
+     *
+     * @param actionRegistry The feature action registry.
+     * @param dataGroupController The data group controller.
+     */
+    public FeatureActionStateController(FeatureActionsRegistry actionRegistry, DataGroupController dataGroupController)
+    {
+        myRegistry = actionRegistry;
+        myDataGroupController = dataGroupController;
+    }
+
+    @Override
+    public void activateState(String id, String description, Collection<? extends String> tags, Node node)
+    {
+        try
+        {
+            NodeList featureActionNodes = StateXML.getChildNodes(node, "/:state/:featureActions/:featureAction");
+            for (int i = 0; i < featureActionNodes.getLength(); i++)
+            {
+                FeatureAction newFeatureAction = new FeatureAction();
+                Node featureActionNode = featureActionNodes.item(i);
+                try
+                {
+                    newFeatureAction = XMLUtilities.readXMLObject(featureActionNode, FeatureAction.class);
+                }
+                catch (JAXBException e)
+                {
+                    LOGGER.error(e.getMessage(), e);
+                }
+                String layerPath = StateXML.newXPath().evaluate("/" + ModuleStateController.STATE_QNAME
+                        + "/:featureActions/:featureAction[@type=\"type\"]", featureActionNode);
```

Comment:
I would recommend making a FeatureActionStateConstants class with these string values (similar to CSVStateConstants, StateConstants, etc.)